queue/driver/driver_test.go

```diff
@@ -93,6 +93,21 @@ func (s *DriverSuite) TestInitialValues() {
 	s.Equal(0, stats.Total)
 }
 
+func (s *DriverSuite) TestPutJobDoesNotAllowDuplicateIds() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	s.NoError(s.driver.Open(ctx))
+
+	j := job.NewShellJob("echo foo", "")
```

Comment:
nit: This tests adding identical jobs, not adding jobs with duplicate IDs, which is a superset. Construct two jobs that differ _except_ in their IDs to test duplicate IDs adds fail.